packages/did/src/DidDetails/FullDidDetails.ts

```diff
@@ -6,178 +6,191 @@
-    this.keyRelationships.none = []
-    const keysWithRelationship = new Set<string>(
-      Array.prototype.concat(...Object.values(keyRelationships))
-    )
-    this.keys.forEach((_, keyId) => {
-      if (!keysWithRelationship.has(keyId)) {
-        this.keyRelationships.none?.push(keyId)
-      }
-    })
-  }
-
-  /**
-   * Gets the next nonce/transaction index required for DID authorized blockchain transactions.
-   *
-   * @param increment Flag indicating whether the retrieved tx index should be increased.
-   * @returns A [[BN]] indicating the next transaction index.
-   */
-  public getNextTxIndex(increment = true): BN {
-    const nextIndex = this.lastTxIndex.addn(1)
-    if (increment) this.lastTxIndex = nextIndex
-    return nextIndex
   }
 
-  /**
-   * Returns all the DID keys that could be used to authorize the submission of the provided call.
-   *
-   * @param call The call to submit.
-   * @returns The set of keys that could be used to sign the call.
-   */
-  public getKeysForCall(call: CallMeta): IDidKeyDetails[] {
-    return getKeysForCall(this, call)
+  public getKeysForExtrinsic(extrinsic: Extrinsic): DidKey[] {
+    const callMethod = extrinsic.method
+    const { section, method } = callMethod
+    const keyRelationship =
+      methodMapping[section][method] ||
+      methodMapping[section].default ||
+      methodMapping.default.default
+    return keyRelationship === 'paymentAccount'
+      ? []
+      : this.getKeys(keyRelationship)
   }
 
-  /**
-   * Returns all the DID keys that could be used to authorize the submission of the provided extrinsic.
-   *
-   * @param apiOrMetadata The node runtime information to use to retrieve the required information.
-   * @param extrinsic The extrinsic to submit.
-   * @returns The set of keys that could be used to sign the extrinsic.
-   */
-  public getKeysForExtrinsic(
-    apiOrMetadata: ApiOrMetadata,
-    extrinsic: Extrinsic
-  ): IDidKeyDetails[] {
-    return getKeysForExtrinsic(apiOrMetadata, this, extrinsic)
+  public async getNextNonce(): Promise<BN> {
+    const currentNonce = await queryNonce(this.identifier)
+    // Wrap around the max u64 value when reached.
+    // FIXME: can we do better than this? Maybe we could expose an RPC function for this, to keep it consistent over time.
+    return currentNonce === maxNonceValue ? new BN(0) : currentNonce.addn(1)
   }
 
-  /**
-   * Signs and returns the provided unsigned extrinsic with the right DID key, if present. Otherwise, it will return an error.
-   *
-   * @param extrinsic The unsigned extrinsic to sign.
-   * @param signer The keystore to be used to sign the encoded extrinsic.
-   * @param submitterAccount The KILT account to bind the DID operation to (to avoid MitM and replay attacks).
-   * @param incrementTxIndex Flag indicating whether the DID nonce should be increased before submitting the operation or not.
-   * @returns The DID-signed submittable extrinsic.
-   */
```

Comment:
Is this jsdoc comment already on the interface? If not I would say that these are quite helpful and should be added again.

Reply:
Yes, I never added them again, so thanks for the heads up -> https://github.com/KILTprotocol/sdk-js/pull/447/commits/717e627262f495ebc47334416e6893bcc1243d80